app/Providers/FortifyServiceProvider.php

```diff
@@ -104,7 +104,7 @@ public function boot(): void
         Fortify::resetUserPasswordsUsing(ResetUserPassword::class);
 
         Fortify::loginView(function () {
-            if (!session()->has('intended_url')) {
+            if (!session()->has('url.intended')) {
```

Comment:
This solution seems to work with hard navigations, but surfaces another bug if navigating from an anchor tag in page.

If I add "auth" middleware to linkedhashes.php, like:

```php
Route::get('linkedhashes.php', fn () => $this->handlePageRequest('linkedhashes'))->middleware(['auth']);
```

and then update the conditional to be like:

```php
if (url()->previous() !== route('login')) {
    session()->put('url.intended', url()->previous());
}
```

Then if I click the "Supported Game Files" link from `/game/1` while unauthenticated, `dump(session()->get('url.intended') outputs "/game/1", and that's where the user is then redirected to upon authenticating.

It sounds like based on this comment https://github.com/RetroAchievements/RAWeb/pull/2231#discussion_r1480570916 we need a new middleware to set the `url.intended` value correctly. Does that sound right, or am I misinterpreting?

Reply:
Interesting, the Framework already should handle this just like that. I'll have another look to see if this behaviour is intended.

That said, I think it will be much cleaner and easier to reason about redirects as soon as we got rid of legacy pages and have auth middlewares on "normal" controllers in place.